packages/core/src/internal/fetchSource.ts

```diff
@@ -172,7 +172,25 @@ async function* fetchOperation(
     } else if (!/text\//i.test(contentType)) {
       results = parseJSON(response);
     } else {
-      throw new Error(await response.text());
+      if (contentType === 'text/plain') {
+        const text = await response.text();
+        if (text.startsWith('{')) {
+          try {
+            results = JSON.parse(text);
```

Comment:
My generator foo is lacking here, does this count, because `yield JSON.parse()` seems invalid